core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala

```diff
@@ -352,6 +539,54 @@ case class MultiClassificationMetricsTopK
   Error: Seq[Double]
 ) extends EvaluationMetrics
 
+/**
+ * Metrics for multi-class confusion matrix. It captures confusion matrix of records of which
+ * 1) the labels belong to the top n most occurring classes (n = confMatrixNumClasses)
+ * 2) the top predicted probability exceeds a certain threshold in confMatrixThresholds
+ *
+ * @param confMatrixNumClasses value of the top n most occurring classes in the dataset
+ * @param confMatrixClassIndices label index of the top n most occuring classes
+ * @param confMatrixThresholds a sequence of thresholds
+ * @param confMatrices a Matrix object that stores the confusion matrix for each threshold in confMatrixThresholds
+ */
+case class MulticlassConfMatrixMetricsByThreshold
```

Comment:
1. Prefixing everything with `contMatrix` seems redundant 
2. Why is everything lowercased here? Why not:
```scala
case class MulticlassConfMatrixMetricsByThreshold
(
  NumClasses: Int,
  ClassIndices: Seq[Double],
  Thresholds: Seq[Double],
  ConfMatrices: Seq[Seq[Long]]
) extends EvaluationMetrics
```

Reply:
1. There's already a `thresholds` defined for the threshold metrics, so adding a `confMatrix` prefix is necessary to avoid confusion. Also `confMatrixNumClasses` and `confMatrixClassIndices` specifically refer to the cutoff classes for confusion matrix calculation, so I think it's better to be explicit about them. Otherwise `numClasses` could mean number of classes for other metrics or for all the data.

2. Yes I can change all the element to uppercase. There doesn't seem to be a consistency of whether to use uppercase or camel case in the case classes that extends `EvaluationMetrics` here.

Reply:
@tovbinm  Changed case class data fields name to upper case. Please see updated PR